static/examples/batch/batch-slicing-threads.scd

```diff
@@ -0,0 +1,86 @@
+
+    slices.loadToFloatArray(action:{ |slicesArr| // Bring the values in the slices buffer from the server to the language as a float array
+        slicesArr.postln;
+        slicesArr.doAdjacentPairs({ |start, end, sliceIndex|
+            /*
+            Take each of the adjacent pairs and pass them to this function as an array of 2 values
+
+            For example, [0,1,2,3,4] will execute this function 4 times, passing these 2 value arrays:
+            [0,1]
+            [1,2]
+            [2,3]
+            [3,4]
+
+            This will give us the left and right boundary of each segment
+            We can use this to inform the analysis (FluidBufSpectralShape) where to start analysing and how many frames to analyse
+            */
+
+            fork ({
+                var sliceAnalysis = Buffer(s); // A Buffer() for writing the analysis from FluidSpectralShape into
+                var sliceCentroid = Buffer(s); // A Buffer() for extracting the centroid into
+                var sliceStats = Buffer(s, 7); // A Buffer() for writing the statistical analyses of the centroid into
+                var sliceMean = Buffer(s, 1); // A Buffer() for extracting just the mean from the 7 statistical analyses.
+
+                var numsamps = end-start; // The next slice point minus the current one will determine how many frames to analyse.
+
+                // Analyse the drum buffer starting at `start` and for a number of samples equal to `numsamps`
+                /*
+                This returns a buffer (sliceAnalysis) that is 7 channels wide (for the 7 spectral analyses) and
+                however many frames long as there are fft frames in the slice.
+                */
+                FluidBufSpectralShape.process(s, source:drum, startFrame:start, numFrames:numsamps, features:sliceAnalysis).wait;
+
+
+                // We extract only the first channel of the 7 channels (the centroid) and put it into a Buffer() named sliceCentroid
+                FluidBufSelect.process(s, source:sliceAnalysis, destination:sliceCentroid, channels:0).wait;
+
+                // Then calculate the statistical analyses of the centroid for each segment.
+                // This will return write the output into a new Buffer() called sliceStats that will be 1 channel and 7 frames in size.
+                FluidBufStats.process(s, source:sliceCentroid, stats:sliceStats).wait;
+
+                // Then extract only the first frame from the 7 frames and write it into a Buffer() called sliceMean.
+                // This isolates a singular value, the average spectral centroid for the segment.
+                FluidBufSelect.process(s, source:sliceStats, destination:sliceMean, indices:0).wait;
+
+                // Now that we have a 1x1 Buffer() holding the mean centroid for that segment we can add it to the FluidDataSet.
+                data.addPoint(sliceIndex, sliceMean);
+                s.sync;
+                counter = counter + 1;
```

Comment:
We could just track the `size` of the DataSet instead (which implies a further server call, but is perhaps clearer in terms of expressing intent)

Reply:
I like that over tracking with a counter.

Reply:
This seems to be problematic likely due to my coding... it always prints the size to the console which confuses what is going on. Is there a way to nab `data.size` without it hassling the user?

Reply:
IIRC, the default action for `size` is a `postln`, so you'd want to override that to move into a local var, which perhaps means you can get rid of the sync and stuff

```js
data.addPoint(sliceIndex, sliceMean);
data.size(action: {|dataSetSize| 
  if(dataSetSize == wav.size, { processCompleted.unhang; }); 
}); 
```